File: src/JVM/label/label.java
package codegen.JVM.label;

import compilers.LineOfCode;
import codegen.JVM.JVMlabel;
import codegen.JVM.JVMgenGlobals;
import codegen.JVM.JVMgenUtils;

public class label implements JVMlabel, LineOfCode{
  private String l;

  public label(){
    l = JVMgenGlobals.LABELTAG +
         JVMgenUtils.getUniqueInt();
  }

  public label(String s) { l = s; }

  public String getLabel() { return l; }

  public String toString() { return l; }
}

File: src/JVM/label/label.cpp
#include <iostream>
#include "JVMlabel.h"
#include "label.h"
#include "JVMlineOfCode.h"
#include "JVMgenGlobals.h"
#include "JVMgenUtils.h"

namespace toyc {
  static int _labelInt = 0;
  
  label::label(){
    l += (LABELTAG + std::to_string(_labelInt++));
  }

  label::label(std::string s) { l = s; }

  std::string label::getLabel() { return l; }

  std::string label::toString() { return l; }
}
